libraries/conf.rb

```diff
@@ -98,7 +98,7 @@ def whyrun_supported?
         execute "#{r.name}.#{r.conf_type}-systemd-reload" do
           command 'systemctl daemon-reload'
           action :nothing
-          only_if { r.is_a?(Chef::Resource::SystemdUnit) }
+          only_if { r.is_a?(Chef::Resource::SystemdUnit) && r.auto_reload }
           subscribes :run, "file[#{conf_path}]", :immediately
```

Comment:
yeah, the common use-case is to require changes be applied immediately, so that's what the default does (auto_reload defaults to true).

but, for folks scaling their unit count into a range where they're willing to accept delays in order to minimize daemon-reload performance hits, we provide an escape hatch (auto_reload false, with optional set_properties action to apply (a subset of accepted) configs immediately).

i like the idea of using a handler to do the post-converge triggering of a daemon-reload if needed. a handler concept definitely seem like the right entry point for this kind of thing, but i don't think this use-case actually fits into the existing handler categories (exception, report, start). it seems like we need a 4th handler category for `end` that runs right at the end of a chef run and still has the ability for exceptions to make it into post-run exception/report handlers.